### DISCUSSION TITLE
Why doesn't `log <path>` respect `revsets.log`?

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#4433] [Q&A]
`log <path>` restricts the list of commits to ones that modified the specified path(s). If `-r` is also given, the effect seems to be equivalent to appending `& file(<path>)` to the explicit revset argument.

But if `-r` is not provided, it appears that the revset used is just `file(<path>)`. Why isn't it `<revsets.log> & file(<path>)`, for consistency (and convenience)? (And of course the user could still use `log <path> -r ::` to get the current behavior.)

Relatedly, is there a way to explicitly use my configured `revsets.log` inside of a revset expression?

**Comment by martinvonz** [comment]
That's actually how it used to behave. It changed in #3410 and there's some discussion there. I was actually surprised myself when `jj log README.md` showed up empty because I had not edited in any commits in my default revset.

**Reply by BatmanAoD** [reply]
Huh, I don't actually see any rationale in that PR...and I see that the ticket was actually something I opened.

The new default actually seems much *more* surprising to me. Now, in a repo that's quite old, I get pages and pages of results when I expect just the changes from my branch. (My configured default revset is `trunk()..@ | trunk()`.)

**Reply by martinvonz** [reply]
It's possible that just printing a warning as you suggested somewhere would be a better solution. I'm not sure. I'd like to hear some more opinions before we change it again.

**Reply by amiryal** [reply]
> I'd like to hear some more opinions before we change it again.

Here :wave: thinking this new behaviour is confusingly inconsistent, although I haven’t experienced it first hand yet. My answer to #2475 would be: don’t change the output, but rather just vary the warning messages depending on whether `X` seems like a filename, a revset or both.
